backend/events/apps.py

```diff
@@ -40,53 +40,82 @@ def get_event_loop():
         loop.close()
 
 
-def tuple_get_worker(tuple_type, _tuple):
-    if tuple_type == 'aggregatetuple':
+def tuple_get_worker(event_type, _tuple):
+    if event_type == 'aggregatetuple':
         return _tuple['worker']
     return _tuple['dataset']['worker']
 
 
-def on_tuples(cc_event, block_number, tx_id, tx_status):
-    payload = json.loads(cc_event['payload'])
+def on_tuples(tx_status, event_type, _tuple):
+
     owner = get_owner()
     worker_queue = f"{LEDGER['name']}.worker"
 
-    for tuple_type, _tuples in payload.items():
+    key = _tuple['key']
+    status = _tuple['status']
+
+    if tx_status != 'VALID':
+        logger.error(
+            f'Failed transaction on task {key}: type={event_type}'
+            f' status={status} with tx status: {tx_status}')
+        return
+
+    logger.info(f'Processing task {key}: type={event_type} status={status}')
+
+    if status != 'todo':
+        return
+
+    if event_type is None:
+        return
+
+    tuple_owner = tuple_get_worker(event_type, _tuple)
+
+    if tuple_owner != owner:
+        logger.debug(f'Skipping task {key}: owner does not match'
+                     f' ({tuple_owner} vs {owner})')
+        return
+
+    if AsyncResult(key).state != 'PENDING':
+        logger.info(f'Skipping task {key}: already exists')
+        return
+
+    prepare_tuple.apply_async(
+        (_tuple, event_type),
+        task_id=key,
+        queue=worker_queue
+    )
+
+
+def on_compute_plan(tx_status, _tuple):
+
+    worker_queue = f"{LEDGER['name']}.worker"
+
+    key = _tuple['computePlanID']
+    status = _tuple['status']
+
+    logger.info(f'Processing cleaning task {key}: type=computePlan status={status}')
+
+    clean_compute_plan.apply_async(
+        (_tuple, ),
+        task_id=key,
+        queue=worker_queue
+    )
+
+
+def on_event(cc_event, block_number, tx_id, tx_status):
+    payload = json.loads(cc_event['payload'])
+
+    for event_type, _tuples in payload.items():
+
         if not _tuples:
             continue
 
         for _tuple in _tuples:
```

Comment:
I think I would prefer not calling it `_tuples` and `_ tuple`, here but `_asset(s)` or something like that.

Reply:
Yes

Reply:
Done